server.go

```diff
@@ -303,6 +300,8 @@ type Server struct {
 	DecorateReader DecorateReader
 	// DecorateWriter is optional, allows customization of the process that writes raw DNS messages.
 	DecorateWriter DecorateWriter
+	// Maximum number of TCP queries before we close the socket (unlimited if 0).
+	MaxTCPQueries int
```

Comment:
I question whether having the default value mean unlimited is best, because it amounts to a behaviour change that could be unexpected. Perhaps it would be best to have 0 mean the current max of 128 and -1 explicitly mean unlimited (maybe even with a `const UnlimitedMaxQueries = -1` for clarity). @miekg thougts?